### DISCUSSION TITLE
Add status enabled/disabled field to user

### DISCUSSION THREAD

**Discussion by maneljuarez** [spatie/laravel-permission#2449] [Q&A]
How can i add a status (enabled/disabled) field to user model to don't allow user login or anything else when it's disabled ?

**Comment by drbyte** [comment]
> How can i add a status (enabled/disabled) field to user model to don't allow user login or anything else when it's disabled ?

I would just do that in application code, not using this package. Just add a `boolean` field (perhaps `'prevent_login'`) to your Users model and table, and then during login check whether that's been set.

If you wanted to use this package for it, you could perhaps add a permission named "prevent login", and check for that during login.

Either way, a couple ways to check one of the above during login would include:
- could create a middleware to check for it, and then add that middleware to the routes that deal with login.
- if using the `laravel/ui` package, override the `attemptLogin()` method by cloning it from `AuthenticatesUsers` trait into `LoginController` and adding a check for it there

Those are some ideas. I think Jeffrey Way covered doing this sort of thing in his various Laracasts videos on Laravel From Scratch and on building custom login authentication.
